### PR TITLE
Add Geneset docs

### PR DESCRIPTION
**Pull request by JureZmrzlikar**

### CODE REVIEW

File: docs/geneset.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+.. note::
+
+  Performing these operations is only possible on genesets that have equal values
+  of ``species`` and ``source`` attribute. Otherwise newly created sets would not
+  make sense and would be inconsistent.
+
+So far, geneset ``gs3`` only exists locally. One can easily save it to Resolwe server::
+
+    gs3.save()
```
Comment (mzganec) [inline docs/geneset.rst:52]: Neat!

File: docs/geneset.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    # Union
+    gs3 = gs1 | gs2
+    # Intersection
+    gs3 = gs1 & gs2
+    # Difference
+    gs3 = gs1 - gs2
+
+.. note::
+
+  Performing these operations is only possible on genesets that have equal values
```
Comment (mzganec) [inline docs/geneset.rst:46]: Well caught.

**Review by mzganec — APPROVED** [review]
A great sibling to the collection tables. The derivation from `Data` and direct support of `set` operations look well thought out.

File: src/resdk/resources/geneset.py [hunk 3]
```diff
@@ -157,9 +157,15 @@ def set_operator(self, operator, other):
         :param other: Geneset object
         :return: new Geneset object
         """
+        # Make sure that self._genes is populated:
+        _ = self.genes
```
Comment (AGregorc) [inline src/resdk/resources/geneset.py:161]: Nice check 👍

**Review by AGregorc — APPROVED** [review]
Nicely explained geneset docs, lgtm

**Pull request merged**